File: 2023/advent_of_code/test/q16/q16_test.exs
defmodule AdventOfCode.Q16Test do
  use ExUnit.Case
  doctest AdventOfCode.Q16

  test "Q16 Part 1 Example works" do
    file_stream = File.stream!("./test/q16/q16_example.txt")
    result = AdventOfCode.Q16.part1(file_stream)
    assert result == 46
  end

  test "Q16 Part 1 works" do
    file_stream = File.stream!("./test/q16/q16.txt")
    result = AdventOfCode.Q16.part1(file_stream)
    assert result == 7111
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q16/Q16Test.kt
package q16

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q16Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q16.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(1641, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q16.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(2261, part2(filePath!!))
    }
}